crates/compiler/can/src/suffixed.rs

```diff
@@ -170,6 +170,41 @@ pub fn unwrap_suffixed_expression<'a>(
                 }
             }
 
+            Expr::Expect(condition, continuation) => {
+                if is_expr_suffixed(&condition.value) {
+                    // we cannot unwrap a suffixed expression within expect
+                    // e.g. expect (foo! "bar")
+                    return Err(EUnwrapped::Malformed);
```

Comment:
Just a note for the future, eventually we'll want to support this when we allow doing `Task` things inside `expect`, but for right now this seems like the best UX!